packages/api/game/types.go

```diff
@@ -0,0 +1,140 @@
+package game
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/websocket"
+)
+
+type BaseMessage struct {
+	Event string          `json:"event"`
+	Data  json.RawMessage `json:"data"`
+}
+
+type PlayerJoinData struct {
+	name string
+	x    float32
+	y    float32
+}
+
+type PlayerMoveData struct {
+	id string
+	x  float32
+	y  float32
+}
+
+type IdData struct {
+	id string
+}
+
+type ItemEntityCreateData struct {
+	item string
+	x    float32
+	y    float32
+}
+
+type Game struct {
+	ctx          context.Context
+	connection   *websocket.Conn
+	players      []*Player
+	itemEntities []*ItemEntity
+}
+
+func (g *Game) HandlePlayerJoin(data PlayerJoinData) {
+	fmt.Println("player join")
```

Comment:
once the context is propagated to these methods, you can get a log instance using

```
ctx, log := base.LogFor(ctx)
```

and use that for logging